# Analyzing Used Car Listings

In this project, we will clean the data and analyze used car listings from ebay.

Data source: https://www.kaggle.com/datasets/sijovm/used-cars-data-from-ebay-kleinanzeigen?resource=download

In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt

In [2]:
autos = pd.read_csv('Desktop/autos.csv', encoding='Latin-1')
autos.info()
autos.head()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 371528 entries, 0 to 371527
Data columns (total 20 columns):
 #   Column               Non-Null Count   Dtype 
---  ------               --------------   ----- 
 0   dateCrawled          371528 non-null  object
 1   name                 371528 non-null  object
 2   seller               371528 non-null  object
 3   offerType            371528 non-null  object
 4   price                371528 non-null  int64 
 5   abtest               371528 non-null  object
 6   vehicleType          333659 non-null  object
 7   yearOfRegistration   371528 non-null  int64 
 8   gearbox              351319 non-null  object
 9   powerPS              371528 non-null  int64 
 10  model                351044 non-null  object
 11  kilometer            371528 non-null  int64 
 12  monthOfRegistration  371528 non-null  int64 
 13  fuelType             338142 non-null  object
 14  brand                371528 non-null  object
 15  notRepairedDamage    299468 non-nu

dateCrawled                            name  seller offerType  \
0  2016-03-24 11:52:17                      Golf_3_1.6  privat   Angebot   
1  2016-03-24 10:58:45            A5_Sportback_2.7_Tdi  privat   Angebot   
2  2016-03-14 12:52:21  Jeep_Grand_Cherokee_"Overland"  privat   Angebot   
3  2016-03-17 16:54:04              GOLF_4_1_4__3TÜRER  privat   Angebot   
4  2016-03-31 17:25:20  Skoda_Fabia_1.4_TDI_PD_Classic  privat   Angebot   

   price abtest vehicleType  yearOfRegistration    gearbox  powerPS  model  \
0    480   test         NaN                1993    manuell        0   golf   
1  18300   test       coupe                2011    manuell      190    NaN   
2   9800   test         suv                2004  automatik      163  grand   
3   1500   test  kleinwagen                2001    manuell       75   golf   
4   3600   test  kleinwagen                2008    manuell       69  fabia   

   kilometer  monthOfRegistration fuelType       brand notRepairedDamage  \
0     150000                    0   benzin  volkswagen               NaN   
1     125000                    5   diesel        audi                ja   
2     125000                    8   diesel        jeep               NaN   
3     150000                    6   benzin  volkswagen              nein   
4      90000                    7   diesel       skoda              nein   

           dateCreated  nrOfPictures  postalCode             lastSeen  
0  2016-03-24 00:00:00             0       70435  2016-04-07 03:16:57  
1  2016-03-24 00:00:00             0       66954  2016-04-07 01:46:50  
2  2016-03-14 00:00:00             0       90480  2016-04-05 12:47:46  
3  2016-03-17 00:00:00             0       91074  2016-03-17 17:40:17  
4  2016-03-31 00:00:00             0       60437  2016-04-06 10:17:21

## Clean columns

In [3]:
# Check column names
autos.columns

Index(['dateCrawled', 'name', 'seller', 'offerType', 'price', 'abtest',
       'vehicleType', 'yearOfRegistration', 'gearbox', 'powerPS', 'model',
       'kilometer', 'monthOfRegistration', 'fuelType', 'brand',
       'notRepairedDamage', 'dateCreated', 'nrOfPictures', 'postalCode',
       'lastSeen'],
      dtype='object')

In [4]:
# Change from camelcase to snakecase
autos.columns = ['date_crawled', 'name', 'seller', 'offer_type', 'price', 'ab_test',
       'vehicle_type', 'registration_year', 'gearbox', 'power_ps', 'model',
       'kilometer', 'registration_month', 'fuel_type', 'brand',
       'unrepaired_damage', 'ad_created', 'num_photos', 'postal_code',
       'last_seen']
autos.head()

date_crawled                            name  seller offer_type  \
0  2016-03-24 11:52:17                      Golf_3_1.6  privat    Angebot   
1  2016-03-24 10:58:45            A5_Sportback_2.7_Tdi  privat    Angebot   
2  2016-03-14 12:52:21  Jeep_Grand_Cherokee_"Overland"  privat    Angebot   
3  2016-03-17 16:54:04              GOLF_4_1_4__3TÜRER  privat    Angebot   
4  2016-03-31 17:25:20  Skoda_Fabia_1.4_TDI_PD_Classic  privat    Angebot   

   price ab_test vehicle_type  registration_year    gearbox  power_ps  model  \
0    480    test          NaN               1993    manuell         0   golf   
1  18300    test        coupe               2011    manuell       190    NaN   
2   9800    test          suv               2004  automatik       163  grand   
3   1500    test   kleinwagen               2001    manuell        75   golf   
4   3600    test   kleinwagen               2008    manuell        69  fabia   

   kilometer  registration_month fuel_type       brand unrepaired_damage  \
0     150000                   0    benzin  volkswagen               NaN   
1     125000                   5    diesel        audi                ja   
2     125000                   8    diesel        jeep               NaN   
3     150000                   6    benzin  volkswagen              nein   
4      90000                   7    diesel       skoda              nein   

            ad_created  num_photos  postal_code            last_seen  
0  2016-03-24 00:00:00           0        70435  2016-04-07 03:16:57  
1  2016-03-24 00:00:00           0        66954  2016-04-07 01:46:50  
2  2016-03-14 00:00:00           0        90480  2016-04-05 12:47:46  
3  2016-03-17 00:00:00           0        91074  2016-03-17 17:40:17  
4  2016-03-31 00:00:00           0        60437  2016-04-06 10:17:21

In [5]:
autos.describe(include='all')

date_crawled         name  seller offer_type         price  \
count                371528       371528  371528     371528  3.715280e+05   
unique               280500       233531       2          2           NaN   
top     2016-03-24 14:49:47  Ford_Fiesta  privat    Angebot           NaN   
freq                      7          657  371525     371516           NaN   
mean                    NaN          NaN     NaN        NaN  1.729514e+04   
std                     NaN          NaN     NaN        NaN  3.587954e+06   
min                     NaN          NaN     NaN        NaN  0.000000e+00   
25%                     NaN          NaN     NaN        NaN  1.150000e+03   
50%                     NaN          NaN     NaN        NaN  2.950000e+03   
75%                     NaN          NaN     NaN        NaN  7.200000e+03   
max                     NaN          NaN     NaN        NaN  2.147484e+09   

       ab_test vehicle_type  registration_year  gearbox       power_ps  \
count   371528       333659      371528.000000   351319  371528.000000   
unique       2            8                NaN        2            NaN   
top       test    limousine                NaN  manuell            NaN   
freq    192585        95894                NaN   274214            NaN   
mean       NaN          NaN        2004.577997      NaN     115.549477   
std        NaN          NaN          92.866598      NaN     192.139578   
min        NaN          NaN        1000.000000      NaN       0.000000   
25%        NaN          NaN        1999.000000      NaN      70.000000   
50%        NaN          NaN        2003.000000      NaN     105.000000   
75%        NaN          NaN        2008.000000      NaN     150.000000   
max        NaN          NaN        9999.000000      NaN   20000.000000   

         model      kilometer  registration_month fuel_type       brand  \
count   351044  371528.000000       371528.000000    338142      371528   
unique     251            NaN                 NaN         7          40   
top       golf            NaN                 NaN    benzin  volkswagen   
freq     30070            NaN                 NaN    223857       79640   
mean       NaN  125618.688228            5.734445       NaN         NaN   
std        NaN   40112.337051            3.712412       NaN         NaN   
min        NaN    5000.000000            0.000000       NaN         NaN   
25%        NaN  125000.000000            3.000000       NaN         NaN   
50%        NaN  150000.000000            6.000000       NaN         NaN   
75%        NaN  150000.000000            9.000000       NaN         NaN   
max        NaN  150000.000000           12.000000       NaN         NaN   

       unrepaired_damage           ad_created  num_photos   postal_code  \
count             299468               371528    371528.0  371528.00000   
unique                 2                  114         NaN           NaN   
top                 nein  2016-04-03 00:00:00         NaN           NaN   
freq              263182                14450         NaN           NaN   
mean                 NaN                  NaN         0.0   50820.66764   
std                  NaN                  NaN         0.0   25799.08247   
min                  NaN                  NaN         0.0    1067.00000   
25%                  NaN                  NaN         0.0   30459.00000   
50%                  NaN                  NaN         0.0   49610.00000   
75%                  NaN                  NaN         0.0   71546.00000   
max                  NaN                  NaN         0.0   99998.00000   

                  last_seen  
count                371528  
unique               182806  
top     2016-04-07 06:45:59  
freq                     17  
mean                    NaN  
std                     NaN  
min                     NaN  
25%                     NaN  
50%                     NaN  
75%                     NaN  
max                     NaN

## Initial Data Exploration and Cleaning

In [6]:
autos.describe(include='all')

date_crawled         name  seller offer_type         price  \
count                371528       371528  371528     371528  3.715280e+05   
unique               280500       233531       2          2           NaN   
top     2016-03-24 14:49:47  Ford_Fiesta  privat    Angebot           NaN   
freq                      7          657  371525     371516           NaN   
mean                    NaN          NaN     NaN        NaN  1.729514e+04   
std                     NaN          NaN     NaN        NaN  3.587954e+06   
min                     NaN          NaN     NaN        NaN  0.000000e+00   
25%                     NaN          NaN     NaN        NaN  1.150000e+03   
50%                     NaN          NaN     NaN        NaN  2.950000e+03   
75%                     NaN          NaN     NaN        NaN  7.200000e+03   
max                     NaN          NaN     NaN        NaN  2.147484e+09   

       ab_test vehicle_type  registration_year  gearbox       power_ps  \
count   371528       333659      371528.000000   351319  371528.000000   
unique       2            8                NaN        2            NaN   
top       test    limousine                NaN  manuell            NaN   
freq    192585        95894                NaN   274214            NaN   
mean       NaN          NaN        2004.577997      NaN     115.549477   
std        NaN          NaN          92.866598      NaN     192.139578   
min        NaN          NaN        1000.000000      NaN       0.000000   
25%        NaN          NaN        1999.000000      NaN      70.000000   
50%        NaN          NaN        2003.000000      NaN     105.000000   
75%        NaN          NaN        2008.000000      NaN     150.000000   
max        NaN          NaN        9999.000000      NaN   20000.000000   

         model      kilometer  registration_month fuel_type       brand  \
count   351044  371528.000000       371528.000000    338142      371528   
unique     251            NaN                 NaN         7          40   
top       golf            NaN                 NaN    benzin  volkswagen   
freq     30070            NaN                 NaN    223857       79640   
mean       NaN  125618.688228            5.734445       NaN         NaN   
std        NaN   40112.337051            3.712412       NaN         NaN   
min        NaN    5000.000000            0.000000       NaN         NaN   
25%        NaN  125000.000000            3.000000       NaN         NaN   
50%        NaN  150000.000000            6.000000       NaN         NaN   
75%        NaN  150000.000000            9.000000       NaN         NaN   
max        NaN  150000.000000           12.000000       NaN         NaN   

       unrepaired_damage           ad_created  num_photos   postal_code  \
count             299468               371528    371528.0  371528.00000   
unique                 2                  114         NaN           NaN   
top                 nein  2016-04-03 00:00:00         NaN           NaN   
freq              263182                14450         NaN           NaN   
mean                 NaN                  NaN         0.0   50820.66764   
std                  NaN                  NaN         0.0   25799.08247   
min                  NaN                  NaN         0.0    1067.00000   
25%                  NaN                  NaN         0.0   30459.00000   
50%                  NaN                  NaN         0.0   49610.00000   
75%                  NaN                  NaN         0.0   71546.00000   
max                  NaN                  NaN         0.0   99998.00000   

                  last_seen  
count                371528  
unique               182806  
top     2016-04-07 06:45:59  
freq                     17  
mean                    NaN  
std                     NaN  
min                     NaN  
25%                     NaN  
50%                     NaN  
75%                     NaN  
max                     NaN

Initial observations:

1. There are a few columns where all (or almost all) of the values are the same:
- seller
- offer_type
2. The num_photos column looks odd.

In [7]:
autos["num_photos"].value_counts()

num_photos
0    371528
Name: count, dtype: int64

The num_photos column has 0 for every column. We'll drop this column, plus the other two we noted as mostly one value.

In [8]:
autos = autos.drop(["num_photos", "seller", "offer_type"], axis=1)

## Explore Kilometer and Price

Let's explore the kilometer column

In [9]:
autos["kilometer"].value_counts()

kilometer
150000    240797
125000     38067
100000     15920
90000      12523
80000      11053
70000       9773
60000       8669
50000       7615
5000        7069
40000       6376
30000       6041
20000       5676
10000       1949
Name: count, dtype: int64

The values in this field are rounded, which might indicate that sellers had to choose from pre-set options for this field. Additionally, there are more high mileage than low mileage vehicles.

Now, let's explore the price column

In [10]:
print(autos["price"].unique().shape)
print(autos["price"].describe())
autos["price"].value_counts().head(20)

(5597,)
count    3.715280e+05
mean     1.729514e+04
std      3.587954e+06
min      0.000000e+00
25%      1.150000e+03
50%      2.950000e+03
75%      7.200000e+03
max      2.147484e+09
Name: price, dtype: float64


price
0       10778
500      5670
1500     5394
1000     4649
1200     4594
2500     4438
600      3819
3500     3792
800      3784
2000     3432
999      3364
750      3203
650      3150
4500     3053
850      2946
2200     2936
700      2936
1800     2886
900      2874
950      2793
Name: count, dtype: int64

There are 10,778 cars listed with $0 price - We will consider removing these rows since they are only 3% of the entire data set.
The maximum price is two billion dollars, which seems a lot, so let's look at the highest prices further.

In [11]:
autos["price"].value_counts().sort_index(ascending=False).head(20)

price
2147483647     1
99999999      15
99000000       1
74185296       1
32545461       1
27322222       1
14000500       1
12345678       9
11111111      10
10010011       1
10000000       8
9999999        3
3895000        1
3890000        1
2995000        1
2795000        1
1600000        2
1300000        1
1250000        2
1234566        1
Name: count, dtype: int64

There is a sudden jump of price from $3,895,000 to $9,999,999, so we'll remove the rows with prices above $3,895,000.

In [12]:
autos = autos[autos["price"].between(1,3895000)]
autos["price"].describe()

count    3.606980e+05
mean     6.068401e+03
std      1.818741e+04
min      1.000000e+00
25%      1.250000e+03
50%      3.000000e+03
75%      7.490000e+03
max      3.895000e+06
Name: price, dtype: float64

## Explore Date Columns

There are a number of columns with date information:

- date_crawled
- registration_month
- registration_year
- ad_created
- last_seen

We'll explore each of these columns to learn more about the listings.

In [13]:
autos[['date_crawled','ad_created','last_seen']][0:5]

date_crawled           ad_created            last_seen
0  2016-03-24 11:52:17  2016-03-24 00:00:00  2016-04-07 03:16:57
1  2016-03-24 10:58:45  2016-03-24 00:00:00  2016-04-07 01:46:50
2  2016-03-14 12:52:21  2016-03-14 00:00:00  2016-04-05 12:47:46
3  2016-03-17 16:54:04  2016-03-17 00:00:00  2016-03-17 17:40:17
4  2016-03-31 17:25:20  2016-03-31 00:00:00  2016-04-06 10:17:21

In [14]:
# See the range and frequency of date_crawled sorted by dates
(autos["date_crawled"]
        .str[:10]
        .value_counts(normalize=True, dropna=False)
        .sort_index()
        )

date_crawled
2016-03-05    0.025545
2016-03-06    0.014480
2016-03-07    0.035653
2016-03-08    0.033466
2016-03-09    0.034112
2016-03-10    0.032648
2016-03-11    0.032775
2016-03-12    0.036244
2016-03-13    0.015781
2016-03-14    0.036327
2016-03-15    0.033424
2016-03-16    0.030200
2016-03-17    0.031655
2016-03-18    0.013119
2016-03-19    0.035268
2016-03-20    0.036399
2016-03-21    0.035695
2016-03-22    0.032493
2016-03-23    0.032002
2016-03-24    0.029914
2016-03-25    0.032798
2016-03-26    0.031969
2016-03-27    0.030230
2016-03-28    0.035060
2016-03-29    0.034123
2016-03-30    0.033529
2016-03-31    0.031874
2016-04-01    0.034150
2016-04-02    0.035096
2016-04-03    0.038811
2016-04-04    0.037630
2016-04-05    0.012786
2016-04-06    0.003127
2016-04-07    0.001619
Name: proportion, dtype: float64

The the site seems to have been crawled for about a one month period; from March 3, 2016 to April 6, 2016 

In [15]:
# See the frequency of dates sorted by ascending values
(autos["date_crawled"]
        .str[:10]
        .value_counts(normalize=True, dropna=False)
        .sort_values()
        )

date_crawled
2016-04-07    0.001619
2016-04-06    0.003127
2016-04-05    0.012786
2016-03-18    0.013119
2016-03-06    0.014480
2016-03-13    0.015781
2016-03-05    0.025545
2016-03-24    0.029914
2016-03-16    0.030200
2016-03-27    0.030230
2016-03-17    0.031655
2016-03-31    0.031874
2016-03-26    0.031969
2016-03-23    0.032002
2016-03-22    0.032493
2016-03-10    0.032648
2016-03-11    0.032775
2016-03-25    0.032798
2016-03-15    0.033424
2016-03-08    0.033466
2016-03-30    0.033529
2016-03-09    0.034112
2016-03-29    0.034123
2016-04-01    0.034150
2016-03-28    0.035060
2016-04-02    0.035096
2016-03-19    0.035268
2016-03-07    0.035653
2016-03-21    0.035695
2016-03-12    0.036244
2016-03-14    0.036327
2016-03-20    0.036399
2016-04-04    0.037630
2016-04-03    0.038811
Name: proportion, dtype: float64

The crawl dates have an evened out frequency distribution

In [16]:
(autos["last_seen"]
        .str[:10]
        .value_counts(normalize=True, dropna=False)
        .sort_index()
        )

last_seen
2016-03-05    0.001264
2016-03-06    0.004098
2016-03-07    0.005201
2016-03-08    0.007940
2016-03-09    0.009823
2016-03-10    0.011458
2016-03-11    0.012953
2016-03-12    0.023241
2016-03-13    0.008409
2016-03-14    0.012174
2016-03-15    0.016327
2016-03-16    0.016415
2016-03-17    0.028703
2016-03-18    0.006892
2016-03-19    0.016332
2016-03-20    0.019889
2016-03-21    0.020031
2016-03-22    0.020510
2016-03-23    0.018018
2016-03-24    0.019163
2016-03-25    0.018997
2016-03-26    0.015955
2016-03-27    0.016720
2016-03-28    0.022190
2016-03-29    0.023285
2016-03-30    0.023721
2016-03-31    0.024239
2016-04-01    0.023898
2016-04-02    0.024971
2016-04-03    0.025320
2016-04-04    0.025531
2016-04-05    0.126968
2016-04-06    0.218942
2016-04-07    0.130422
Name: proportion, dtype: float64

The crawler recorded the date it last saw any listing, which allows us to determine on what day a listing was removed, presumably because the car was sold.

The last three days contain a disproportionate amount of 'last seen' values. Given that these are 6-10x the values from the previous days, it's unlikely that there was a massive spike in sales, and more likely that these values are to do with the crawling period ending and don't indicate car sales.

In [17]:
print(autos["ad_created"].str[:10].unique().shape)
(autos["ad_created"]
        .str[:10]
        .value_counts(normalize=True, dropna=False)
        .sort_index()
        )

(114,)


ad_created
2014-03-10    0.000003
2015-03-20    0.000003
2015-06-11    0.000003
2015-06-18    0.000003
2015-08-07    0.000003
                ...   
2016-04-03    0.038999
2016-04-04    0.037738
2016-04-05    0.011619
2016-04-06    0.003119
2016-04-07    0.001555
Name: proportion, Length: 114, dtype: float64

There is a large variety of ad created dates. Most fall within 1-2 months of the listing date, but a few are quite old, with the oldest at almost a year ago from the first date the site was crawled.

In [18]:
autos["registration_year"].describe()

count    360698.000000
mean       2004.451877
std          82.110038
min        1000.000000
25%        1999.000000
50%        2004.000000
75%        2008.000000
max        9999.000000
Name: registration_year, dtype: float64

The year that the car was first registered will likely indicate the age of the car. Looking at this column, we note some odd values. The minimum value is 1000, long before cars were invented and the maximum is 9999, many years into the future.

## Dealing with Incorrect Registration Year Data

Because a car can't be first registered before the listing was seen, any vehicle with a registration year above 2016 is definitely inaccurate. Determining the earliest valid year is more difficult. Realistically, it could be somewhere in the first few decades of the 1900s.

One option is to remove the listings with these values. Let's determine what percentage of our data has invalid values in this column:

In [19]:
(~autos["registration_year"].between(1900,2016)).sum() / autos.shape[0]

0.03875264071328369

Since this is less than 4% of our data, we will remove these rows

In [20]:
# Select rows in a dataframe that fall within a value range for a column.
autos = autos[autos["registration_year"].between(1900,2016)]

In [21]:
# Check the frequency of registration year
autos["registration_year"].value_counts(normalize=True).head(10).sort_index()

registration_year
1998    0.049691
1999    0.063544
2000    0.066702
2001    0.056948
2002    0.054280
2003    0.056547
2004    0.056169
2005    0.062670
2006    0.057704
2007    0.050490
Name: proportion, dtype: float64

It appears that most of the vehicles have registration years from 1998-2007.

## Exploring Price by Brand

In [22]:
# Check frequency of listings sorted by brand
autos["brand"].value_counts(normalize=True)

brand
volkswagen        0.211675
bmw               0.109870
opel              0.106397
mercedes_benz     0.096833
audi              0.089536
ford              0.068912
renault           0.047508
peugeot           0.030148
fiat              0.025686
seat              0.018658
skoda             0.015684
mazda             0.015381
smart             0.014329
citroen           0.013948
nissan            0.013596
toyota            0.012930
hyundai           0.009971
sonstige_autos    0.009547
mini              0.009382
volvo             0.009146
mitsubishi        0.008234
honda             0.007531
kia               0.006913
suzuki            0.006365
alfa_romeo        0.006308
porsche           0.006259
chevrolet         0.005024
chrysler          0.003862
dacia             0.002495
jeep              0.002195
land_rover        0.002166
daihatsu          0.002160
subaru            0.002117
jaguar            0.001736
saab              0.001465
daewoo            0.001457
trabant           0.00

German brands represent the top five brands, almost 60% of the overall listings. Volkswagen is by far the most popular brand, with approximately double the cars for sale of the next two brands combined.

There are lots of brands that don't have a significant percentage of listings, so we will limit our analysis to brands representing more than 5% of total listings.

In [23]:
# Filter out brands with less than 5% of total listings
brand_counts = autos["brand"].value_counts(normalize=True)
common_brands = brand_counts[brand_counts > .05].index
print(common_brands)

Index(['volkswagen', 'bmw', 'opel', 'mercedes_benz', 'audi', 'ford'], dtype='object', name='brand')


Now let's get the average prices for each brand representing more than 5% of total listings

In [24]:
# Create list of lists
brand_mean_prices = {}

for brand in common_brands:
    # From the autos dataset, create a list of car listings for each unique brand
    brand_only = autos[autos["brand"] == brand]
    # Get the average price of each brand
    mean_price = brand_only["price"].mean()
    # Turn the average price into an integer
    brand_mean_prices[brand] = int(mean_price)

brand_mean_prices

{'volkswagen': 5452,
 'bmw': 8573,
 'opel': 3028,
 'mercedes_benz': 8609,
 'audi': 9166,
 'ford': 3779}

Of the top 5 brands, there is a distinct price gap:

Audi, BMW and Mercedes Benz are more expensive
Ford and Opel are less expensive
Volkswagen is in between - this may explain its popularity, it may be a 'best of 'both worlds' option.

## Exploring Mileage

In [25]:
# Turn brand_mean_prices into a data frame
bmp_series = pd.Series(brand_mean_prices)
pd.DataFrame(bmp_series, columns=["mean_price"])

mean_price
volkswagen           5452
bmw                  8573
opel                 3028
mercedes_benz        8609
audi                 9166
ford                 3779

In [26]:
brand_mean_mileage = {}

for brand in common_brands:
    # From the autos dataset, create a list of car listings for each unique brand
    brand_only = autos[autos["brand"] == brand]
    # Get the average kilometers of each brand
    mean_mileage = brand_only["kilometer"].mean()
    # Turn the average price into an integer
    brand_mean_mileage[brand] = int(mean_mileage)

mean_mileage = pd.Series(brand_mean_mileage).sort_values(ascending=False)
mean_prices = pd.Series(brand_mean_prices).sort_values(ascending=False)

In [27]:
brand_info = pd.DataFrame(mean_mileage,columns=['mean_mileage'])
brand_info

mean_mileage
bmw                  132791
mercedes_benz        130566
audi                 129440
opel                 128721
volkswagen           128387
ford                 123663

In [28]:
# Add average prices by brand to the data frame
brand_info["mean_price"] = mean_prices
brand_info

mean_mileage  mean_price
bmw                  132791        8573
mercedes_benz        130566        8609
audi                 129440        9166
opel                 128721        3028
volkswagen           128387        5452
ford                 123663        3779

The range of car mileages does not vary as much as the prices do by brand, instead all falling within 10% for the top brands. There is a slight trend to the more expensive vehicles having higher mileage, with the less expensive vehicles having lower mileage.